ahiv/kafka/internal/tcpconnection.h

```diff
@@ -84,21 +84,26 @@ class TCPConnection : public uvw::Emitter<TCPConnection> {
     this->once<E>(listener);
   }
 
-  // Write data into the stream
-  void Write(protocol::Buffer& buffer,
-             const std::function<void(protocol::Buffer&)> responseCallback) {
-    int32_t correlationId = this->idCounter.fetch_add(1);
-    this->responseCallbacks.emplace(ResponseCorrelationCallback{
-      correlationId : correlationId,
-      responseCallback : responseCallback
-    });
-
-    buffer.Overwrite<int32_t>(8, correlationId);
-    handle->write(buffer.Data(), buffer.Size());
+  // Send will serialize a packet, transmit it over TCP and deserialize its
+  // response packet and call the given callback
+  template <typename Message>
+  void Send(typename Message::Request& request,
+            ahiv::kafka::ResponseCallback<typename Message::Response>
+                responseCallback) {
+    ahiv::kafka::protocol::Buffer requestBuffer;
+    requestBuffer.EnsureAllocated(request.Size());
+    request.Write(requestBuffer);
+
```

Comment:
If you have empty lines like this in your method, you should often try to break the method up into smaller methods.